networking/metallb/metallb-configure-return-traffic.adoc

```diff
@@ -0,0 +1,49 @@
+:_content-type: ASSEMBLY
+[id="metallb-configure-return-traffic"]
+= Managing symmetric routing with MetalLB
+include::_attributes/common-attributes.adoc[]
+:context: metallb-configure-return-traffic
+
+toc::[]
+
+As a cluster administrator, you can effectively manage traffic for pods behind a MetalLB load-balancer service with multiple host interfaces by implementing features from MetalLB, NMState and OVN-Kubernetes. By combining these features in this context, you can provide symmetric routing, traffic segregation, and support clients on different networks with overlapping CIDR addresses. 
+
+To achieve this functionality, learn how to implement virtual routing and forwarding (VRF) instances with MetalLB, and configure egress services.
+
+:FeatureName: Configuring symmetric traffic by using VRF instance with MetalLB and an egress service
+include::snippets/technology-preview.adoc[]
+
+== Challenges of managing symmetric routing with MetalLB 
+
+When you use MetalLB with multiple host interfaces, MetalLB exposes and announces a service through all available interfaces on the host. This can present challenges relating to network isolation, asymmetric return traffic and overlapping CIDR addresses. 
+
+One option to ensure that return traffic reaches the correct client is to use static routes. However, MetalLB cannot isolate the services and then announce each service through a different interface. Additionally, static routing requires manual configuration and can add additional network administration if a node shuts down. 
+
+Another challenge emerges in scenarios where MetalLB assigns the IP address of the host network interface as the source IP address for applications running on pods behind a MetalLB service. This is problematic because some external systems expect the source and destination IP address for an application to be the same.
+
+== Overview of managing symmetric routing using VRF-based networks with MetalLB
+
+You can overcome the challenges of implementing symmetric routing with MetalLB by using NMState to configure a VRF instance on a host, associating the VRF instance with MetalLB and configuring an egress service for egress traffic with OVN-Kubernetes.
+
+The configuration process involves three stages:
+
+.1. Define a VRF and routing rules
+
+* Configure a `NodeNetworkConfigurationPolicy` custom resource (CR) to associate a VRF instance with a network interface. 
+* Use the VRF's routing table to direct ingress and egress traffic.
+
+.2. Link the VRF to a MetalLB `BGPPeer`
+
+* Configure a MetalLB `BGPPeer` resource to use the VRF instance on a network interface. 
+* By associating the `BGPPeer` resource with the VRF instance, the designated network interface becomes the primary interface for the BGP session, and MetalLB advertises the services through this interface.
+
+.3. Configure an egress service
+
+* Configure an ergress service to use the MetalLB service IP address as the source IP for egress traffic.
```

Comment:
@fedepaol  ^^ Are we setting the MetalLB service IP address as the source IP address for egress traffic in this full workflow? We don't have the `sourceIPBy` value set?

Reply:
this is probably a leftover from the original version (before we had the sourceIPBy field), I assume it should be LoadBalancerIP here, but I'll let Federico answer

Reply:
Ah yes I think the `sourceIPBy: "LoadBalancerIP"` field was missing from the full workflow in the gDoc so I didn't include it in the EgressService CR. But this probably from an earlier implementation. I've added `sourceIPBy: "LoadBalancerIP"` now in the EgressService CR in the full workflow

Reply:
The egress service serves two purpouses:
- to pick the network
- to force the source ip to be the lb ip

The first is required in this scenario, the latter is optional (so I would add this as 3-a or something like that), "if you want the traffic coming from pods to be"...